src/index.ts

```diff
@@ -149,11 +191,30 @@ export { extractPublicKey } from './utils.js'
  */
 const signLegacyV1 = async (privateKey: PrivateKey, value: Uint8Array, validityType: IpnsEntry.ValidityType, validity: Uint8Array): Promise<Uint8Array> => {
   try {
-    const dataForSignature = ipnsEntryDataForV1Sig(value, validityType, validity)
+    const dataForSignature = ipnsRecordDataForV1Sig(value, validityType, validity)
 
     return await privateKey.sign(dataForSignature)
   } catch (error: any) {
     log.error('record signature creation failed', error)
     throw errCode(new Error('record signature creation failed'), ERRORS.ERR_SIGNATURE_CREATION)
   }
 }
+
+/**
+ * Normalizes the given record value. It ensures it is a string starting with '/'.
+ * If the given value is a cid, the returned path will be '/ipfs/{cid}'.
+ */
+const normalizeValue = (value: string | Uint8Array): string => {
+  const str = typeof value === 'string' ? value : uint8ArrayToString(value)
+
+  if (str.startsWith('/')) {
```

Comment:
I think more validation needs to be done here, as written '/' will be accepted, as will '/foo', '/foo/bar/baz', etc.

Reply:
A path of `/` probably isn't valid?

Reply:
Depends: it would error if IPNS name is resolved by IPFS Gateway, but there could be JS app that uses IPNS records for publishing application-specific paths, and in that use case  `/`  or `/foo` could be ok. 

(namely, we have CBOR `Data` field where people can now put arbitrary data, so users may end up putting noop  `/` in the `Value` field)